source/portable/os/ota_os_posix.c

```diff
@@ -150,23 +175,46 @@ OtaOsStatus_t Posix_OtaReceiveEvent( OtaEventContext_t * pEventCtx,
     OtaOsStatus_t otaOsStatus = OtaOsSuccess;
     char * pDst = pEventMsg;
     char buff[ MAX_MSG_SIZE ];
+    struct timespec ts = { 0 };
+    time_t currentTime = time( NULL );
 
     ( void ) pEventCtx;
     ( void ) timeout;
 
     /* Receive the next event from OTA event queue.*/
     errno = 0;
 
-    if( mq_receive( otaEventQueue, buff, sizeof( buff ), NULL ) == -1 )
+    /* Set receive timeout. */
+    ts.tv_nsec = MS_TO_NS( MS_LESS_THAN_S( timeout ) );
+    ts.tv_sec = MS_TO_S( timeout );
+
+    /* Detect overflow. */
+    if( ( currentTime > 0 ) && ( ( int64_t ) ( INT32_MAX - ts.tv_sec ) >= ( int64_t ) currentTime ) )
+    {
+        ts.tv_sec += currentTime;
+    }
+    else
+    {
+        ts.tv_sec = INT32_MAX;
```

Comment:
On most recent systems, time_t is 64 bits. On linux this is defined by the [TIME_T_MAX macro](https://elixir.bootlin.com/linux/v3.3/source/include/linux/time.h#L43). 

On linux, timespec_add_safe is available which could be a helpful reference.

Reply:
No need to calculate absolute time because poll uses relarive time.